prespective_rectification/abrupt_corners.py

```diff
@@ -0,0 +1,131 @@
+
+
+def select_points_by_segment(point_start, point_end, len_points):
+    if point_start > point_end:
+        moved_initial_point = len_points - point_start
+        mid_point = (moved_initial_point + point_end) // 2
+        difference = int((point_end - point_start) * 0.25)
+        marked_point_lower = mid_point - difference - moved_initial_point
+        marked_point_upper = mid_point + difference - moved_initial_point
+        if marked_point_lower < 0:
+            marked_point_lower = len_points + marked_point_lower
+    else:
+        mid_point = (point_start + point_end) // 2
+        difference = int((point_end - point_start) * 0.25)
+        marked_point_lower = mid_point - difference
+        marked_point_upper = mid_point + difference
+    return marked_point_lower, marked_point_upper
+
+
+def get_corners_abrupt_changes(cumulative_difference):
+    flag_off = True
+    total_cumulative = []
+    for i, element in enumerate(cumulative_difference):
+
+        if not element:
+
+            if flag_off:
+                current_cumulative = []
+                counter = 0
+
+                for series_contigous in cumulative_difference[i:]:
+                    if series_contigous:
+                        break
+                    counter += 1
+
+                    current_cumulative.append(i + counter)
+                total_cumulative.append(current_cumulative)
+                flag_off = False
+        else:
+            flag_off = True
+    total_cumulative = sorted(total_cumulative, key=lambda l: (len(l), l))
+    quadratic_changes = []
+
+    for change in total_cumulative[-2:]:
+        quadratic_changes.extend([change[0], change[len(change) - 1]])
+    quadratic_changes.sort()
+
+    return quadratic_changes
+
+
+def abrupt_changes_algorithm(image, masked_image):
+    # gather and clean contours
+    cnts = cv2.findContours(masked_image.astype("uint8"), cv2.RETR_EXTERNAL,
+                            1)
+
+    cnts = imutils.grab_contours(cnts)
+
+    cleaned_cnts = []
+    for point_element in cnts[0]:
+        cleaned_cnts.append(list(point_element))
+    plot_points(cleaned_cnts, image)
+
+    # project the contours in a sum of diference list
+    previous_element = None
+    list_diference_x, list_diference_y = [], []
+    diference_sum_x, diference_sum_y = 0, 0
+    for element in cnts[0]:
+        if previous_element is not None:
+            diference_sum_x += (element[0][0] - previous_element[0][0])
+            list_diference_x.append(diference_sum_x)
+            diference_sum_y += (element[0][1] - previous_element[0][1])
+            list_diference_y.append(diference_sum_y)
+        previous_element = element
+    y_range = range(len(list_diference_x))
+
+    smoothed_difference_y = savgol_filter(list_diference_y, 13, 5)  # window size 13, polynomial order 5
+    smoothed_difference_x = savgol_filter(list_diference_x, 13, 5)  # window size 13, polynomial order 5
```

Comment:
You could also use named arguments instead of adding the comment behind the sentence.

Reply:
makes sense